client/lobby/SelectionTab.cpp

```diff
@@ -111,10 +111,32 @@ bool mapSorter::operator()(const std::shared_ptr<CMapInfo> aaa, const std::share
 	}
 }
 
+// pick sorting order based on selection
+static ESortBy SortBySelectionScreen(ESelectionScreen Type)
+{
+	// initialize 'generalSortingBy' early
+	switch(Type)
+	{
+	case ESelectionScreen::newGame:
+		return ESortBy::_name;
+	case ESelectionScreen::loadGame:
+	case ESelectionScreen::saveGame:
+		return ESortBy::_fileName;
+	case ESelectionScreen::campaignList:
+		return ESortBy::_name;
+	}
+	// should not reach here
+	assert(0);
```

Comment:
I'd rather return _name in all the rest cases and remove assert at all. It is definitely not the situation where you want game to crash.

Reply:
Sounds good. Changed to return `_name`.